### DISCUSSION TITLE
Next-Auth with credentials returns always 401

### DISCUSSION THREAD

**Discussion by fred-10git** [nextauthjs/next-auth#5385] [Help]
I'm trying to add authentication on my Next.JS (localhost:3000) app with credentials from a Strapi API (localhost:2337) with NextAuth.js (each app is on a docker container).

First of all, i've written a common signIn function :

```
export const signInWithCredentials = async ({ email, password }) => {
    const fetch_url = "http://localhost:2337/api/auth/local";
    const params = {
      method: "POST",
      mode: "cors",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        identifier: email,
        password,
      }),
    };

    try {
        const response = await fetch(fetch_url, params);
        const data = await response.json();
        const user = { ...data.user, jwt: data.jwt }
        return user;
    } catch (err) {
        console.log(err);  // HERE I get a 401 ERROR when using through Next-Auth
    }
    return user || null;
}
```

I made a test page on my next.js app, and it's working, i can get my strapi user through this function without error.
But that's not working while trying the next-auth login with credentials.

My [..nextauth.js] file :

```
export default NextAuth({
  providers: [
    CredentialsProvider({
      name: "credentials",
      credentials: {
        email: { label: "Email", type: "text", placeholder: "email@domain.com" },
        password: { label: "Mot de passe", type: "password" },
      },
      authorize: async (credentials, req) => {
        const user = await signInWithCredentials(credentials);
        return user;
      },
    }),
  ],
  secret: conf.nextAuthSecret,
 }
```

But when I try to login (from a custom login page or through the generated page http://localhost:3000/api/auth/signin), I always got a 401 unauthorized exception in my signInWithCredentials  function 

If someone could explain what i'm missing ...

**Comment by usmanity** [comment]
One thing I noticed is in your `signInWithCredentials` function, you're passing "identifier" instead of "email". Could this be causing an issue?
```
      body: JSON.stringify({
        identifier: email,
        password,
      }),
```

**Reply by fred-10git** [reply]
Hi!
No, `identifier ` is what Strapi API is waiting (and the same function is working called independantly).
